src/Payments/Api.php

```diff
@@ -80,8 +80,12 @@ public function add_payment_method_script_dependencies( $dependencies, $handle )
 	 * Add payment method data to Asset Registry.
 	 */
 	public function add_payment_method_script_data() {
-		$script_data = $this->payment_method_registry->get_all_registered_script_data();
+		// Enqueue the order of enabled gateways as `paymentGatewayOrder`.
+		$available_gateways = WC()->payment_gateways->get_available_payment_gateways();
+		$this->asset_registry->add( 'paymentGatewayOrder', array_keys( $available_gateways ) );
 
+		// Enqueue all registered gateway data (settings/config etc).
+		$script_data = $this->payment_method_registry->get_all_registered_script_data();
```

Comment:
We could potentially tweak the structure of the data to combine order and config. So each gateway's script data is would be nested under the gateway id, in a single key on wcSettings. For example:

```js
wcSettings [ 
   bacs: {
      accountName: 'Nike.com bank account', 
      ...
   },
   stripe: {
   } 
]
```

In this PR I've just added a new key for the order to minimise code impact :)